keps/sig-auth/20191101-add-request-id-to-k8s-logs.md

```diff
@@ -0,0 +1,322 @@
+---
+title: Add Request-ID to each Kubernetes component log
+authors:
+ - "@hase1128"
+ - "@sshukun"
+ - "@furukawa3"
+ - "@vanou"
+owning-sig: sig-instrumentation
+participating-sigs:
+reviewers:
+ - TBD
+approvers:
+ - TBD
+editor: TBD
+creation-date: 2019-11-01
+last-updated: 2020-03-17
+status: provisional
+---
+
+# Add Request-ID to each Kubernetes component log
+
+## Table of Contents
+
+<!-- toc -->
+ - [Summary](#summary)
+ - [Motivation](#motivation)
+   - [Target User](#target-user)
+   - [Target User's objective](#target-users-objective)
+   - [Case 1](#case-2)
+   - [Case 2](#case-1)
+   - [Goals](#goals)
+   - [Non-Goals](#non-goals)
+ - [Proposal](#proposal)
+   - [Basic policy of Request-ID](#basic-policy-of-request-id)
+   - [Design Overview](#design-overview)
+     - [Design of Propagate Request-ID](#design-of-propagate-request-id)
+     - [Design of Export Request-ID](#design-of-export-request-id)
+     - [Design overview of Control Request-ID](#design-overview-of-control-request-id)
+     - [Detail design of Control Request-ID](#detail-design-of-control-request-id)
+ - [Test Plan](#test-plan)
+ - [Migration / Graduation Criteria](#migration--graduation-criteria)
+   - [Alpha](#alpha)
+   - [Beta](#beta)
+   - [GA](#ga)
+
+<!-- /toc -->
+
+## Summary
+
+This KEP proposes a new unique logging meta-data into all Kubernetes logs. It makes us
+more easy to identify specific logs related to a single user operation (such as
+`kubectl apply -f <my-pod.yaml>`). This feature is similar to
+[Global request ID](https://docs.openstack.org/api-guide/compute/faults.html) for
+OpenStack. It greatly reduces investigation cost.
+
+## Motivation
+
+### Target User
+
+Support team in k8s Service Provider
+
+### Target User's objective
+
+We'd like to resolve quickly for end users' problem.
+
+Tracking logs among each Kubernetes component related to specific an user operation is very tough work. It is necessary to match logs by basically using timestamps and object's name as hints. If multiple users throw many API requests at the same time, it is very difficult to track logs across each Kubernetes component log. 
+
+It is difficult that support team in k8s Service Provider resolve end user's problem quickly in the above. Therefore, we'd like to add a new identifier which is unique to each user operation. This feature is useful for the following use cases:
+
+#### Case 1
```

Comment:
It would help to readers to give these cases summaries. This one would probably be something like:
**Given an API Request, find the resulting component logs**

The one below would be something like
**Given a component log, find the API Request**

Reply:
I added cases summaries.